lib/ansible/modules/utilities/logic/wait_for_connection.py

```diff
@@ -92,6 +94,30 @@
 
   - name: Gather facts for first time
     setup:
+
+# Build an online instance, wait for it to become ready and continue (playbook
+# runs against localhost).
+- hosts: localhost
+  tasks:
+    - name: Create instance on Google Cloud
+      gce:
+        name: instance-name
+        zone: europe-west1-b
+        network: default
+        subnet: subnet-name
+        machine-type: n1-highcpu-4
+        image-family: windows-2016
+      register: instance
+    - add_host: &windows
+        hostname: win_instance
+        ansible_host: "{{ instance.instance_data[0].public_ip }}"
+        ansible_connection: winrm
+        ansible_port: 5986
+        ansible_winrm_server_cert_validation: ignore
+        ansible_winrm_transport: ntl
+        ansible_password: password
+    - wait_for_connection:
+      vars: *windows
```

Comment:
So I don't think this is something we want to expose to users as the ideal way to do this. See my other comment.